From f3ecca39c6ff61e8373d830c3ffc71cd18a85951 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Thu, 23 Jul 2020 10:20:24 -0400
Subject: cygwin: Backport SVN 197170

2013-03-27 Kai Tietz <ktietz@redhat.com>

Add support for cygwin x64 target.

        * dfp.m4: Add support for cygwin x64 target.
        * picflag.m4: Likewise.
---
 config/dfp.m4     | 2 +-
 config/picflag.m4 | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/config/dfp.m4 b/config/dfp.m4
index cc778b19bbb..f03088798b9 100644
--- a/config/dfp.m4
+++ b/config/dfp.m4
@@ -22,7 +22,7 @@ Valid choices are 'yes', 'bid', 'dpd', and 'no'.]) ;;
   case $1 in
     powerpc*-*-linux* | i?86*-*-linux* | x86_64*-*-linux* | s390*-*-linux* | \
     i?86*-*-mingw* | x86_64*-*-mingw* | \
-    i?86*-*-cygwin*)
+    i?86*-*-cygwin* | x86_64*-*-cygwin*)
       enable_decimal_float=yes
       ;;
     *)
diff --git a/config/picflag.m4 b/config/picflag.m4
index bd818125a66..2ee5cd0785f 100644
--- a/config/picflag.m4
+++ b/config/picflag.m4
@@ -17,7 +17,9 @@ case "${$2}" in
     hppa*64*-*-hpux*)
 	# PIC is the default for 64-bit PA HP-UX.
 	;;
-    i[[34567]]86-*-cygwin* | i[[34567]]86-*-mingw* | x86_64-*-mingw*)
+    i[[34567]]86-*-cygwin* | x86_64-*-cygwin*)
+	;;
+    i[[34567]]86-*-mingw* | x86_64-*-mingw*)
 	;;
     i[[34567]]86-*-interix[[3-9]]*)
 	# Interix 3.x gcc -fpic/-fPIC options generate broken code.
-- 
